src/algokit/core/tasks/ipfs.py

```diff
@@ -1,111 +1,118 @@
+import json
 import logging
-from collections.abc import Generator
 from pathlib import Path
 
 import httpx
 import keyring
 
 logger = logging.getLogger(__name__)
 
-ALGOKIT_WEB3_STORAGE_NAMESPACE = "algokit_web3_storage"
-ALGOKIT_WEB3_STORAGE_TOKEN_KEY = "algokit_web3_storage_access_token"
+ALGOKIT_PINATA_NAMESPACE = "algokit_pinata"
+ALGOKIT_PINATA_TOKEN_KEY = "algokit_pinata_access_token"
 
 MAX_FILE_SIZE = 100 * 1024 * 1024  # 100MB
 MAX_CHUNK_SIZE = 1024 * 1024  # 1MB
 DEFAULT_TIMEOUT = 10
 
 
-class Web3StorageError(Exception):
-    """Base class for Web3 Storage errors."""
+class PinataError(Exception):
+    """Base class for Piñata errors."""
 
     def __init__(self, response: httpx.Response):
         self.response = response
-        super().__init__(f"Web3 Storage error: {response.status_code}")
+        super().__init__(f"Pinata error: {response.status_code}")
 
     def __str__(self) -> str:
-        return f"Web3 Storage error: {self.response.status_code}. {self.response.text}"
+        return f"Pinata error: {self.response.status_code}. {self.response.text}"
 
 
-class Web3StorageBadRequestError(Web3StorageError):
+class PinataBadRequestError(PinataError):
     pass
 
 
-class Web3StorageUnauthorizedError(Web3StorageError):
+class PinataUnauthorizedError(PinataError):
     pass
 
 
-class Web3StorageForbiddenError(Web3StorageError):
+class PinataForbiddenError(PinataError):
     pass
 
 
-class Web3StorageInternalServerError(Web3StorageError):
+class PinataInternalServerError(PinataError):
     pass
 
 
-class Web3StorageHttpError(Web3StorageError):
+class PinataHttpError(PinataError):
     pass
 
 
-def get_web3_storage_api_key() -> str | None:
+def get_pinata_jwt() -> str | None:
     """
     Retrieves a password from the keyring library using the
-    ALGOKIT_WEB3_STORAGE_NAMESPACE and ALGOKIT_WEB3_STORAGE_TOKEN_KEY variables.
+    ALGOKIT_PINATA_NAMESPACE and ALGOKIT_PINATA_TOKEN_KEY variables.
 
     Returns:
         str | None: The retrieved password from the keyring, or None if no password is found.
     """
-
-    return keyring.get_password(ALGOKIT_WEB3_STORAGE_NAMESPACE, ALGOKIT_WEB3_STORAGE_TOKEN_KEY)
-
-
-def set_web3_storage_api_key(api_key: str | None) -> None:
+    try:
+        old_api_key = keyring.get_password("algokit_web3_storage", "algokit_web3_storage_access_token")
+        if old_api_key:
+            logger.warning(
+                "You are using the old Web3 Storage API key. Please login again using `algokit ipfs login` with Pinata "
+                "ipfs provider. Follow the instructions on https://docs.pinata.cloud/docs/getting-started "
+                "to create an account and obtain a JWT."
+            )
+            keyring.delete_password("algokit_web3_storage", "algokit_web3_storage_access_token")
+    except KeyError as ex:
+        logger.debug("Old Web3 Storage API key is already deleted.", exc_info=ex)
```

Comment:
This exception will be thrown and caught under normal conditions, so I feel we shouldn't have a debug message here.